geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/CommandIntegrationTest.java

```diff
@@ -0,0 +1,135 @@
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import org.apache.geode.NativeRedisTestRule;
+import org.apache.geode.redis.GeodeRedisServerRule;
+
+public class CommandIntegrationTest {
+
+  @ClassRule
+  public static NativeRedisTestRule redisServer = new NativeRedisTestRule("redis:6.2.4");
+
+  @ClassRule
+  public static GeodeRedisServerRule radishServer = new GeodeRedisServerRule();
+
+  private RedisCommands<String, String> redisClient;
+  private RedisCommands<String, String> radishClient;
+
+  @Before
+  public void setup() {
+    redisClient =
+        RedisClient.create(String.format("redis://%s:%d", BIND_ADDRESS, redisServer.getPort()))
+            .connect().sync();
+
+    radishClient =
+        RedisClient.create(String.format("redis://%s:%d", BIND_ADDRESS, radishServer.getPort()))
+            .connect().sync();
+  }
+
+  @After
+  public void teardown() {}
+
+  @Test
+  public void commandReturnsResultsMatchingNativeRedis() {
+    Map<String, CommandStructure> goldenResults = processRawCommands(redisClient.command());
+    Map<String, CommandStructure> results = processRawCommands(radishClient.command());
+
+    List<String> commands = new ArrayList<>(results.keySet());
+    Collections.sort(commands);
+
+    SoftAssertions softly = new SoftAssertions();
+    for (String command : commands) {
+      softly.assertThatCode(() -> compareCommands(results.get(command), goldenResults.get(command)))
+          .as("command: " + command)
+          .doesNotThrowAnyException();
+    }
+    softly.assertAll();
+  }
+
+  private void compareCommands(CommandStructure actual, CommandStructure expected) {
+    assertThat(actual).as("no metadata for " + expected.name).isNotNull();
+    SoftAssertions softly = new SoftAssertions();
+    softly.assertThat(actual.arity).as(expected.name + ".arity").isEqualTo(expected.arity);
+    softly.assertThat(actual.flags).as(expected.name + ".flags")
+        .containsExactlyInAnyOrderElementsOf(expected.flags);
+    softly.assertThat(actual.firstKey).as(expected.name + ".firstKey").isEqualTo(expected.firstKey);
+    softly.assertThat(actual.lastKey).as(expected.name + ".lastKey").isEqualTo(expected.lastKey);
+    softly.assertThat(actual.stepCount).as(expected.name + ".stepCount")
+        .isEqualTo(expected.stepCount);
+    softly.assertAll();
+  }
+
+  @SuppressWarnings("unchecked")
+  private Map<String, CommandStructure> processRawCommands(List<Object> rawCommands) {
+    Map<String, CommandStructure> commands = new HashMap<>();
+
+    for (Object rawEntry : rawCommands) {
+      List<Object> entry = (List<Object>) rawEntry;
+      String key = (String) entry.get(0);
+      List<String> flags = new ArrayList<>();
+      flags.addAll((List<String>) entry.get(2));
+
+      CommandStructure cmd = new CommandStructure(
+          key,
+          (long) entry.get(1),
+          flags,
+          (long) entry.get(3),
+          (long) entry.get(4),
+          (long) entry.get(5));
```

Comment:
This can be simplified slightly to inline the `flags` variable:
```
      CommandStructure cmd = new CommandStructure(
          key,
          (long) entry.get(1),
          (List<String>) entry.get(2),
          (long) entry.get(3),
          (long) entry.get(4),
          (long) entry.get(5));
```

Reply:
Done